api-tests/core/upload/content-api/graphql-upload.test.api.js

```diff
@@ -425,4 +429,45 @@ describe('Upload plugin end to end tests', () => {
       },
     });
   });
+
+  test('Returns an error when required headers for csrf protection are missing', async () => {
+    const formData = {
+      operations: JSON.stringify({
+        query: /* GraphQL */ `
+          mutation uploadFile($file: Upload!) {
+            upload(file: $file) {
+              data {
+                id
+                attributes {
+                  name
+                  mime
+                  url
+                }
+              }
+            }
+          }
+        `,
+        variables: {
+          file: null,
+        },
+      }),
+      map: JSON.stringify({
+        nFile1: ['variables.file'],
+      }),
+      nFile1: fs.createReadStream(path.join(__dirname, '../utils/rec.jpg')),
+    };
+
+    const res = await rq({ method: 'POST', url: '/graphql', formData, headers: {} });
+
+    expect(res.statusCode).toBe(400);
+    expect(res.body.errors).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({
```

Comment:
Should we add a more precise check to make sure it's related to CSRF, or do you think it's enough?